### DISCUSSION TITLE
Sync Tabs and their order to another Firefox Profile / Browser

### DISCUSSION THREAD

**Discussion by davidjhi** [mbnuqw/sidebery#1488] [General]
UseCase:

- Multiple computers, all with different Firefox Profiles (Same Firefox Account).
- All the sessions are showed in Sidebery -> Settings -> Sync. But can only syncronize Settings, Styles, Context and Keybindings.
- It would be perfect to also **syncronize open tabs and their order in multiple computers.**

In my personal use case, i have +50 permanent pinned tabs, that i would like to have in all my computers / sessions to develop muscle memory with mouse& Keybindings.

Any suggestions?. Thanks!

**Comment by mbnuqw** [comment]
#54, It's planned but will probably require some third-party (or self-hosted) service because of Firefox [limitations](https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/API/storage/sync#storage_quotas_for_sync_data). Not sure about when it will be implemented, since I don't need that feature myself.

Awkward workarounds for now:
- Save/Convert tab panel to Bookmarks. Sync with default Firefox sync. On another device - in Sidebery -> Bookmarks -> Synced dir -> "Open as tabs panel". Then you will be able to use context menu options on that tab panel: "Restore from bookmarks" and "Save to bookmarks" to sync your tabs with bookmarks, which will in turn sync across devices by Firefox sync.
- You can setup auto-export of snapshots to JSON (e.g. in some directory in "Downloads" (it's impossible to select a directory out of "Downloads", which is another Firefox limitation)) and setup your cloud service to sync these files. On another device - you can manually import that snapshot and restore the tabs.

**Reply by hollisticated-horse** [reply]
Cheers for that answer. Could WebDAV be supported, for example ? Quite a few people have access to a Nextcloud account that permits that quite well (also managing bookmarks but besides the point).

And what about doing the odd converting to bookmarks and back to Tabs on both machines through firefox, as an automatic option that is opt-in ? 

By the way, I've let the folks over in [#1372](https://github.com/mbnuqw/sidebery/issues/1372) know about your answer here @mbnuqw.

**Reply by lufixSch** [reply]
Thanks for that answer. I didn't even know that workflow with the bookmarks existed. That's already way better, than my previous workflow.

Wouldn't it be possible to automate the process (e.g. refreshing tab groups from bookmarks and saving tabs groups too bookmarks) every x minutes and/or every time a tab is added or removed?

**Reply by davidjhi** [reply]
Yeah ill be happy to have an option to sync trough a 3rd party, Nextcloud / Drive / DropBox. Hope someday we can make it work, altough the workaround #2 its fine for me at the moment.

**Reply by hollisticated-horse** [reply]
@davidjhi how do you use workaround 2 ? (took me a while to understand the linked issue has nothing to do with syncing :D)

**Reply by rieje** [reply]
Would it be feasible/possible for Sidebery to autoimport a snapshot (which I assume includes grouped tabs), i.e. perhaps just simply use the latest snapshot from a directory always? I already use Syncthing to automatically sync open tabs but it doesn't handle grouped tabs--this could be a solution. Or maybe sync sessionstore.jsonlz4 + Sidebery's `/storage/default/moz-extension+++6ade8aba-e19d-46eb-a8fe-685bdea84655^userContextId=4294967295` directory (if this directory contains all of Sidebery's settings)? Would this directory be named the same in all profiles with Sidebery installed?

Anyway, my approach with syncing tabs in order is to simply have a wrapper script for firefox that copies a profile's sessionstore.jsonlz4 to a sync directory for Syncthing. On starting/stopping Firefox, I copy from/to respectively this file to the location of the local profile. It works well (not sure if pinned tabs are supported) but crucially tree hierarchy is. 

I think tab groups will help further with tab management so I'm interested in a way that includes this support. I could sync the entire Firefox profile with Syncthing but that's significantly more data than a simple sessionstore.jsonlz4 file.

**Reply by directentis1** [reply]
For the first solution, I thought if it were possible to automate this process (listen for changes from the specified sync bookmark folder - when there is a sync event from Firefox Sync) and automatically import them into the Tab Panel specified by Sidebery.

**Comment by davidjhi** [comment]
As i mention [here](https://github.com/mbnuqw/sidebery/issues/1372), i would encourage people to donate, so the developer might consider this as something that could really help us!

![imagen](https://github.com/user-attachments/assets/8ddd3dc7-1541-4c5a-bd10-471cddd3a89d)
